source/uwp/SharedRenderer/lib/Util.cpp

```diff
@@ -657,3 +657,71 @@ std::string ExtractSvgDataFromUri(winrt::Windows::Foundation::Uri const& imageUr
     }
     return data;
 }
+
+winrt::Windows::Foundation::Size ParseSizeOfSVGImageFromXmlString(winrt::hstring const& content)
+{
+    // Parse the size from the XamlDocument as XML
+    winrt::XmlDocument xmlDoc;
+
+    xmlDoc.LoadXml(content);
+
+    if (xmlDoc)
+    {
+        auto rootElement = xmlDoc.DocumentElement();
+
+        // Root element must be an SVG
+        if (rootElement.NodeName() == L"svg")
+        {
+            auto heightAttribute = rootElement.GetAttribute(L"height");
+            auto widthAttribute = rootElement.GetAttribute(L"width");
+
+            double height{0.0};
+            double width{0.0};
+
+            if (!heightAttribute.empty())
+            {
+                height = TryHStringToDouble(heightAttribute).value_or(0.0);
+            }
+
+            if (!widthAttribute.empty())
+            {
+                width = TryHStringToDouble(widthAttribute).value_or(0.0);
+            }
+
+            return {static_cast<float>(width), static_cast<float>(height)};
+        }
+    }
+
+    return {};
+}
+
+winrt::IAsyncOperation<winrt::Windows::Foundation::Size> ParseSizeOfSVGImageFromStreamAsync(winrt::IRandomAccessStream const stream)
+{
+    auto inputStream = stream.GetInputStreamAt(0);
+    auto dataReader = winrt::DataReader(inputStream);
+
+    // Load the data from the stream
+    uint32_t numBytesLoaded = co_await dataReader.LoadAsync(static_cast<uint32_t>(stream.Size()));
+
+    // Read the data as a string
+    winrt::hstring svgString = dataReader.ReadString(numBytesLoaded);
+
+    co_return ParseSizeOfSVGImageFromXmlString(svgString);
+}
+
+bool IsSvgImage(winrt::Windows::Foundation::Uri const& imageUrl)
+{
+    auto imagePath = HStringToUTF8(imageUrl.Path());
+
+    // Find the position of the first semicolon
+    size_t semicolonPos = imagePath.find(';');
+
+    // If there is no semicolon, check the entire string
```

Comment:
I am not sure we should ever do this since there is a decent chance of "svg" appearing in the encoded data.

Reply:
@meetikasharma, you probably want to search both ends, from the begin and end, and when you search from the front you can restrict your search to the exact length you are looking for, and the same for the end. But I would definitely look up for the best practice.

Reply:
We need to look in the entire string for urls which don't have semicolon e.g. `"url": "https://adaptivecards.io/content/adaptive-card.svg"`

But yes, even this format could have svg somewhere in there.

@jwoo-msft FYI

Even if I look for `.svg` in the end, the data format could have that too for other image formats, right?

Reply:
We should probably have different detection for svg depending on if it is http? or data.

Reply:
@meetikasharma & @Apxov, Instead of searching for the entire string if ';' not found; we could just search the end for "svg" from the end. current scheme works the problem is the occurrence of "svg" in the middle...

Reply:
It is just trying to test two totally different things at the same time.  It needs test for svg in each scheme differently.

Reply:
I agree. we could modify it to check if data then check the beginning, else check at the end.

Reply:
Updated the method @jwoo-msft @Apxov